pandas/core/generic.py

```diff
@@ -8763,6 +8763,51 @@ def transform(self, func, *args, **kwargs):
         scalar : type of index
         """
 
+    def _find_first_valid(self, reversed=False):
```

Comment:
can you rename to ``_find_valid_index`` and make the param (required), and call it ``how='first'|'last'``

Reply:
cool, looks better.